java/maven/src/org/netbeans/modules/maven/modelcache/MavenProjectCache.java

```diff
@@ -112,6 +118,9 @@ public MavenProject run() {
                             return mp;
                         }
                     }
+                    if (doNotLoadReturnNull) {
```

Comment:
whi is evil ;) 
wouldnt two seperate methods with descriptive names be better: one which loads project and another which tries to load it or somethig?